src/components/CurrencyInput/CurrencyInput.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from 'react-emotion';
+import { withTheme } from 'emotion-theming';
+
+import IconInput from '../IconInput';
+import HtmlElement from '../HtmlElement';
+import { childrenPropType, themePropType } from '../../util/shared-prop-types';
+
+const currencyBaseStyles = ({ theme }) => css`
+  color: ${theme.colors.b700};
+`;
+
+const currencyStyle = ({ theme, className }) =>
+  cx(currencyBaseStyles({ theme }), className);
+
+/**
+ * SearchInput component for forms.
+ */
+const CurrencyInput = ({
+  children,
+  currencyPosition,
+  currency,
+  theme,
+  ...props
+}) => (
+  <IconInput
+    iconLeft={({ className }) =>
+      currencyPosition === 'left' && (
+        <HtmlElement className={currencyStyle({ theme, className })}>
+          {currency}
+        </HtmlElement>
+      )
+    }
+    iconRight={({ className }) =>
+      currencyPosition === 'right' && (
+        <HtmlElement className={currencyStyle({ theme, className })}>
+          {currency}
+        </HtmlElement>
+      )
+    }
+    {...{ theme, ...props }}
+  >
+    {children}
+  </IconInput>
+);
+
+/**
+ * Triggers disabled styles on the component. This is also forwarded as
+ * attribute to the <input> element.
+ */
+CurrencyInput.propTypes = {
+  theme: themePropType.isRequired,
```

Comment:
At some point (I know this is in progress) please add the documentation for these.